### PR TITLE
change in debug.js for better responsiveness

### PR DESCRIPTION
**Pull request by chikoYEAT**
Making the main execution asynchronous, allows for better handling of asynchronous operations (can improve performance) + also simplified variable declarations + in `Byg` configuration object removed `:` for a syntactic improvement

changing source redeclaration to modifiedSource might improve error handling

### CODE REVIEW

File: runner/debug.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ let lastLine;
 let output = '';
 
 try {
-  const { exports } = compile(source, process.argv.includes('--module') ? [ 'module' ] : [], {
+  const { exports } = await compile(source, process.argv.includes('--module') ? [ 'module' ] : [], {
```
Comment (CanadaHonk) [inline runner/debug.js:46]: The function isn't asynchronous?

File: runner/debug.js [hunk 2]
```diff
@@ -111,7 +111,7 @@ try {
     }
   }, s => output += s);
 
-  exports.main();
+  await exports.main();
```
Comment (CanadaHonk) [inline runner/debug.js:114]: Also not asynchronous.

File: runner/debug.js [hunk 3]
```diff
@@ -27,7 +27,7 @@ const breakpoints = new Array(lines.length);
 let paused = true;
 const byg = Byg({
   lines: originalLines,
-  pause: () => { paused = true; },
+  pause() { paused = true; },
```
Comment (CanadaHonk) [inline runner/debug.js:30]: This change is essentially meaningless and a lot of other code in the project follows the previous syntax anyway?

**Review by CanadaHonk — CHANGES_REQUESTED** [review]
Thanks for the PR but essentially all the changes here don't do anything?

File: runner/debug.js (hunk 3 above)
Comment (chikoYEAT) [reply runner/debug.js:30]: `pause() { paused = true; },` uses object shorthand syntax introduced in ES6, being honest this syntax is shorter and cleaner to read also it implements implicit naming - automatically uses property name which can be helpful in debugging

**Comment by chikoYEAT** [comment]
i am sorry, it's my bad - i assuming those function synchronous (idk why)

File: runner/debug.js (hunk 3 above)
Comment (CanadaHonk) [reply runner/debug.js:30]: Yeah but using it in just this one place is largely meaningless and just a change to make a change? (imo no offence)

Comment (chikoYEAT) [reply runner/debug.js:30]: nvm, If you allow me to perform that i can give a PR after changing this on each and every file + some more changes related to ES6.

Comment (CanadaHonk) [reply runner/debug.js:30]: > also it implements implicit naming
`foo: () => { ... }` also does implicit naming.

Thanks, but I don't think the changes are necessary; it just seems like changing to change.

**Pull request closed without merging**